### PR TITLE
AggregateProofType nil when doing batch updates

### PR DESCRIPTION
**Pull request by rvagg**
@magik6k I don't know whether this is helpful or not but I needed this for what I was doing so here it is. It's currently using the branch @ https://github.com/filecoin-project/go-state-types/pull/242. The cbor-gen changes here are immaterial, I just happened to update cbor-gen to master but the code it outputs should be essentially the same and that could even be backed out. The important change is in `storage/pipeline/commit_batch.go`.

I'm just not sure about `ProveReplicaUpdates3Params`, you should probably look at storage/pipeline/states_replica_update.go and follow up on the work you did in #11226 on that?

### CODE REVIEW

**Comment by rvagg** [comment]
Depends on builtin-actors update to v13.0.0-rc.2, which doesn't exist quite yet.

**Review by magik6k — APPROVED** [review]
one nit, technically doesn't matter, but would be less confusing to drop the network version check

File: storage/pipeline/commit_batch.go [hunk 1]
```diff
@@ -387,7 +387,10 @@ func (b *CommitBatcher) processBatchV2(cfg sealiface.Config, sectors []abi.Secto
 		res.Error = err.Error()
 		return []sealiface.CommitBatchRes{res}, xerrors.Errorf("getting aggregate proof type: %w", err)
 	}
-	params.AggregateProofType = arp
+
+	if aggregate || nv < network.Version22 {
+		params.AggregateProofType = &arp // not optional prior to nv22, can't be set when not aggregating after nv22
```
[the remark this answers is not in the document]
Comment (rvagg) [reply storage/pipeline/commit_batch.go]: oh, right, this is only in the `if nv >= MinDDONetworkVersion` path! sorry, resolved

**Comment by rvagg** [comment]
Pulled in the actors update from https://github.com/filecoin-project/lotus/pull/11604 to here to get to green CI before merging. Two birds, one green stone.

**Pull request merged**